src/main/kotlin/org/opensearch/index/translog/ReplicationTranslogDeletionPolicy.kt

```diff
@@ -0,0 +1,107 @@
+package org.opensearch.index.translog
+
+import org.opensearch.common.unit.ByteSizeValue
+import org.opensearch.index.IndexSettings
+import org.opensearch.index.seqno.RetentionLease
+import org.opensearch.index.seqno.RetentionLeases
+import org.opensearch.replication.ReplicationPlugin
+import java.io.IOException
+import java.util.function.Supplier
+
+class ReplicationTranslogDeletionPolicy(
+    indexSettings: IndexSettings,
+    private val retentionLeasesSupplier: Supplier<RetentionLeases>
+) : TranslogDeletionPolicy() {
+    @Volatile
+    private var translogPruningEnabled: Boolean =
+        ReplicationPlugin.REPLICATION_INDEX_TRANSLOG_PRUNING_ENABLED_SETTING.get(indexSettings.settings)
+
+    @Volatile
+    private var retentionSizeInBytes: Long =
+        if (translogPruningEnabled)
+            ReplicationPlugin.REPLICATION_INDEX_TRANSLOG_RETENTION_SIZE.get(indexSettings.settings).bytes
+        else indexSettings.translogRetentionSize.bytes
+
+
+    @Volatile
+    private var retentionAgeInMillis: Long = indexSettings.translogRetentionAge.millis
+
+    @Volatile
+    private var retentionTotalFiles: Int = indexSettings.translogRetentionTotalFiles
+
+    init {
+        indexSettings.scopedSettings.addSettingsUpdateConsumer(
+            ReplicationPlugin.REPLICATION_INDEX_TRANSLOG_PRUNING_ENABLED_SETTING
+        ) { value: Boolean -> translogPruningEnabled = value }
+
+        indexSettings.scopedSettings.addSettingsUpdateConsumer(
+            ReplicationPlugin.REPLICATION_INDEX_TRANSLOG_RETENTION_SIZE
+        ) { value: ByteSizeValue -> if (translogPruningEnabled) retentionSizeInBytes = value.bytes }
```

Comment:
we want to achieve following behavior:
- If translogPruning is enabled
   - index translog retetnion size should have default value of `512MB` and all the set and get operations should work.
- If translogPruning is disabled  
   - index translog retention size should always be `-1` and set operations shouldn't update the value